src/event.rs

```diff
@@ -546,8 +583,27 @@ impl<N: Notify> Processor<N> {
             window.hide();
         }
 
-        if self.window_changes.hide {
-            window.hide();
+        #[cfg(target_os = "macos")]
+        {
+            use glutin::os::macos::WindowExt;
+
+            if self.window_changes.toggle_fullscreen && !self.is_simple_fullscreen {
+                self.set_fullscreen(window, !self.is_fullscreen);
+                self.is_fullscreen = !self.is_fullscreen;
+            }
```

Comment:
Actually, since for `macos` we need to first check `self.is_simple_fullscreen` I think the code is simplest if we just leave it as is.

Open to suggestions though :)